ros2_batch_job/util.py

```diff
@@ -167,12 +167,28 @@ def warn(*args, **kwargs):
 
 class MyProtocol(AsyncSubprocessProtocol):
     def __init__(self, cmd, exit_on_error, *args, **kwargs):
-        self.cmd = cmd
+        self.cmd = cmd  
         self.exit_on_error = exit_on_error
+        self.progress_bar = False
+        self.progress = []
```

Comment:
I'll suggest renaming this something more like `self.progress_data`; that will make it a little clearer on what it is for.